src/test/java/org/assertj/core/api/InstanceOfAssertFactoriesTest.java

```diff
@@ -0,0 +1,105 @@
+package org.assertj.core.api;
+
+import static java.util.Arrays.asList;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.*;
+
+import java.security.Certificate;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+
+import org.assertj.core.util.Strings;
+import org.junit.jupiter.api.Test;
+
+public class InstanceOfAssertFactoriesTest {
```

Comment:
a few general comments for most of the tests:
- use GIVEN WHEN THEN to better separate the data from the action, here we can skip WHEN and simply use WHEN THEN
- declare the actual value as `Object` to avoid the cast when practical (so not for test where the value is lambda)

example for `completable_future_factory_should_allow_completable_future_assertions`:
```java
// GIVEN
Object value = completedFuture("done");
// THEN
assertThat(value).instanceOf(COMPLETABLE_FUTURE).isDone();
```

Reply:
That's also my preferred style but I noticed that was not widely used in the codebase. I'll adapt my tests to it.